packages/chakra-ui-theme/types/index.d.ts

```diff
@@ -0,0 +1,25 @@
+export type Theme = {
+  breakpoints: { [key: string]: string }
+  zIndices: { [key: string]: string | number }
+  radii:    { [key: string]: string }
+  opacity:  { [key: string]: string }
+  borders:  { [key: string]: string }
+  colors:   { [key: string]: string | { [opacity: string]: string }}
+  fonts: {
+    heading: string
+    body: string
+    mono: string
+  }
+  fontSizes:      { [key: string]: string }
+  fontWeights:    { [key: string]: number }
+  letterSpacings: { [key: string]: string }
+  lineHeights:    { [key: string]: string }
+  borderWidths:   { [key: string]: string }
+  shadows:  { [key: string]: string }
+  sizes:    { [key: string]: string }
+  space:    { [key: string]: string }
+}
+
+declare const theme: Theme
```

Comment:
Hey @Pwuts Is it possible that we can derive a deep Partial from the original theme object here? Something like:

```ts
import theme from './src`

export type Theme = DeepPartial<typeof theme>
```

Reply:
I think the definition of Theme shouldn't be partial, but it could be derived from the default theme, sure.